## Import Libraries

In [1]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
import seaborn as sns

## Data Loading

In [4]:
salaries = pd.read_csv('data/ds_salaries.csv').drop(['Unnamed: 0'], axis=1)

In [5]:
salaries.head()

work_year experience_level employment_type                   job_title  \
0       2020               MI              FT              Data Scientist   
1       2020               SE              FT  Machine Learning Scientist   
2       2020               SE              FT           Big Data Engineer   
3       2020               MI              FT        Product Data Analyst   
4       2020               SE              FT   Machine Learning Engineer   

   salary salary_currency  salary_in_usd employee_residence  remote_ratio  \
0   70000             EUR          79833                 DE             0   
1  260000             USD         260000                 JP             0   
2   85000             GBP         109024                 GB            50   
3   20000             USD          20000                 HN             0   
4  150000             USD         150000                 US            50   

  company_location company_size  
0               DE            L  
1               JP            S  
2               GB            M  
3               HN            S  
4               US            L

### Data Characteristics

In [7]:
print(f'Number of rows: {salaries.shape[0]}')
print(f'Number of columns: {salaries.shape[1]}')

Number of rows: 607
Number of columns: 11


In [8]:
salaries.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 607 entries, 0 to 606
Data columns (total 11 columns):
 #   Column              Non-Null Count  Dtype 
---  ------              --------------  ----- 
 0   work_year           607 non-null    int64 
 1   experience_level    607 non-null    object
 2   employment_type     607 non-null    object
 3   job_title           607 non-null    object
 4   salary              607 non-null    int64 
 5   salary_currency     607 non-null    object
 6   salary_in_usd       607 non-null    int64 
 7   employee_residence  607 non-null    object
 8   remote_ratio        607 non-null    int64 
 9   company_location    607 non-null    object
 10  company_size        607 non-null    object
dtypes: int64(4), object(7)
memory usage: 52.3+ KB


From the above information, we can see that the data already pretty clean and **doesn't have any missing values**. The data also has:
- 4 integer columns
- 7 object columns

In [9]:
salaries.describe().T

count           mean           std     min      25%       50%  \
work_year      607.0    2021.405272  6.921330e-01  2020.0   2021.0    2022.0   
salary         607.0  324000.062603  1.544357e+06  4000.0  70000.0  115000.0   
salary_in_usd  607.0  112297.869852  7.095726e+04  2859.0  62726.0  101570.0   
remote_ratio   607.0      70.922570  4.070913e+01     0.0     50.0     100.0   

                    75%         max  
work_year        2022.0      2022.0  
salary         165000.0  30400000.0  
salary_in_usd  150000.0    600000.0  
remote_ratio      100.0       100.0

The dataset contains information about salaries from **2020** to **2022** with the average salary around **$112,298** and the lowest salary is **$2,859** and the highest salary is **$600,000** per year.